bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/WindowsDefenderConfigurator.java

```diff
@@ -108,14 +108,21 @@ public void handleEvent(Event event) {
 
 	private static boolean runStartupCheck() {
 		if (Platform.isRunning() && Platform.OS.isWindows() && !Platform.inDevelopmentMode()) {
-			IProduct product = Platform.getProduct();
-			if (product != null) {
-				return "org.eclipse.ui.ide.workbench".equals(product.getApplication()); //$NON-NLS-1$
-			}
+			return "org.eclipse.ui.ide.workbench".equals(getRunningApplicationId()); //$NON-NLS-1$
 		}
 		return false;
 	}
 
+	private static String getRunningApplicationId() {
+		@SuppressWarnings("restriction")
+		String appId = System.getProperty(org.eclipse.core.internal.runtime.InternalPlatform.PROP_APPLICATION);
+		if (appId != null) {
+			return appId;
+		}
+		IProduct product = Platform.getProduct();
+		return product != null ? product.getApplication() : null;
+	}
+
```

Comment:
Reading the System-properties is effectively what's done in the proposed solution in https://github.com/eclipse-platform/eclipse.platform.ui/pull/1453#issuecomment-1943428622 (`EnvironmentInfo` only also considers a potential SecurityManager before reading the System-properties, but I assume that nobody is using SecurityManagers anymore).
It is set at startup in `org.eclipse.equinox.internal.app.CommandLineArgs` class.

On the long run it might be interesting to have something like this available as API, e.g. `Platform.getApplication()` similar to `Platform.getProduct()` since this might be of interest for others as well.